src/safir/testing/kubernetes.py

```diff
@@ -742,6 +824,426 @@ async def list_namespaced_event(
+        *,
+        field_selector: Optional[str] = None,
+        label_selector: Optional[str] = None,
+        resource_version: Optional[str] = None,
+        timeout_seconds: Optional[int] = None,
+        watch: bool = False,
+        _preload_content: bool = True,
+        _request_timeout: Optional[int] = None,
+    ) -> V1IngressList | Mock:
+        """List ingress objects in a namespace.
+
+        This does support watches.
+
+        Parameters
+        ----------
+        namespace
+            Namespace of ingresss to list.
+        field_selector
+            Only ``metadata.name=...`` is supported. It is parsed to find the
+            ingress name and only ingresss matching that name will be returned.
+        label_selector
+            Which events to retrieve when performing a watch.  All
+            labels must match.
+        resource_version
+            Where to start in the event stream when performing a watch. If
+            `None`, starts with the next change.
+        timeout_seconds
+            How long to return events for before exiting when performing a
+            watch.
+        watch
+            Whether to act as a watch.
+        _preload_content
+            Verified to be `False` when performing a watch.
+        _request_timeout
+            Ignored, accepted for compatibility with the watch API.
+
+        Returns
+        -------
+        kubernetes_asyncio.client.V1IngressList or unittest.mock.Mock
+            List of ingresss in that namespace, when not called as a watch. If
+            called as a watch, returns a mock ``aiohttp.Response`` with a
+            ``readline`` metehod that yields the events.
+
+        Raises
+        ------
+        AssertionError
+            Some other ``field_selector`` was provided.
+        kubernetes_asyncio.client.ApiException
+            Raised with 404 status if the namespace does not exist.
+        """
+        self._maybe_error("list_namespaced_ingress", namespace, field_selector)
+        if namespace not in self._objects:
+            msg = f"Namespace {namespace} not found"
+            raise ApiException(status=404, reason=msg)
+        if not watch:
+            if field_selector:
+                match = re.match(r"metadata\.name=(.*)$", field_selector)
+                assert match and match.group(1)
+                try:
+                    ingress = self._get_object(
+                        namespace, "Ingress", match.group(1)
+                    )
+                    return V1IngressList(kind="Ingress", items=[ingress])
+                except ApiException:
+                    return V1IngressList(kind="Ingress", items=[])
+            else:
+                ingresss = []
+                if "Ingress" in self._objects[namespace]:
+                    for obj in self._objects[namespace]["Ingress"].values():
+                        ingresss.append(obj)
+                return V1IngressList(kind="Ingress", items=ingresss)
```

Comment:
`label_selector` should be supported for the non-watch case as well for consistency.

Reply:
I'm not seeing where `label_selector` is supported for the non-watch case.